### ISSUE TITLE
Checkpoint Loading Hangs for Trained Neural ODE Model

### ISSUE THREAD

**Issue by octadion** [SDML-KU/qkvflow#2]
Hello, thanks for the great work

so i successfully trained a smaller-scale model on Google Colab (using the openwebtext_10k dataset for ~4000 steps with significant reduced parameters).

I am now trying to load the saved checkpoint for analysis, but the process hangs.

Observed Behavior:

1. The process hangs indefinitely after the "Loading checkpoint from..." log message.
2. The execution becomes completely uninterruptible, requiring a full runtime termination.
3. Resource Usage During Hang:
- GPU VRAM quickly loads to a high, stable level (e.g., 11.2 GB / 15 GB).
- However, System RAM remains stable at a very low level (e.g., ~3.4 GB) and does not increase.

Environment:

Platform: Google Colab
levanter: 1.1
jax: 0.4.2

Question:
Is this a known issue for this model architecture? What is the recommended way to load this checkpoint for analysis, given this specific hanging behavior after the model is loaded to the GPU?

Thank you for your time and help.

**Comment by octadion** [comment]
Tried loading the checkpoint on GPUs with 20 GB and 24 GB VRAM. The hanging issue no longer occurs, though I still hit OOM errors. Looks like the problem was insufficient GPU memory. Will test on 32 GB next and report back.

**Comment by anh-tong** [comment]
Hi there,

In my experience, I do not encounter this. I don't think it's the problem from hardwares because loading model require much less resource than trainning (computing gradients consumes much more GPU).  GPU 11.2 /15 is the preallocation mechanism from JAX (https://docs.jax.dev/en/latest/gpu_memory_allocation.html).

You can use the `get_model` from https://github.com/SDML-KU/qkvflow/blob/main/qkvflow/analysis/utils.py .

We can load the model just like the notebook here https://github.com/SDML-KU/qkvflow/blob/main/notebooks/sensitivity_analysis.ipynb

```python
trainer, model, eval_loader, tokenizer = levanter.config.main(
    get_model,
    args=[
        "--config_path",
        config_path,    # your config path
        "--model_choice",
        model_choice,    # neuralode
        "--trainer.load_checkpoint_path",
        checkpoint_path,
        "--trainer.wandb.mode",
        "disabled",
        "--trainer.ray.start_workers",
        "false",
        "--trainer.ray.auto_start_cluster",
        "false",
    ]
    + additional_args,
)()
```

A small note that you might include the following when using Jupyter notebook.
```python
# this is to prevent loading checkpoint error
# from https://github.com/langchain-ai/langchain/issues/9014
import nest_asyncio  # noqa

nest_asyncio.apply()
```
